core/amber/src/main/scala/edu/uci/ics/texera/web/resource/UserDictionaryResource.scala

```diff
@@ -0,0 +1,238 @@
+    * 401 unauthorized - payload: ErrorResponse("Invalid session"),
+    * 400 bad request - payload: ErrorResponse("Malformed JSON payload"),
+    */
+  @POST
+  @Path("/set")
+  def setValue(
+      @Session session: HttpSession,
+      req: PostRequest
+  ): Response = {
+    val user = UserResource.getUser(session).orNull
+    if (user == null)
+      Response.status(Response.Status.UNAUTHORIZED).entity(ErrorResponse("Invalid session")).build()
+    else if (req == null || req.key == null || req.value == null) {
+      Response
+        .status(Response.Status.BAD_REQUEST)
+        .entity(ErrorResponse("Malformed JSON payload"))
+        .build()
+    } else if (dictEntryExists(user, req.key)) {
+      userDictionaryDao.update(new UserDictionary(user.getUid, req.key, req.value))
+      Response.ok(ConfirmationResponse("Updated")).build()
+    } else {
+      userDictionaryDao.insert(new UserDictionary(user.getUid, req.key, req.value))
+      Response.ok(ConfirmationResponse("Created")).build()
+    }
+  }
+
+  /**
+    * This method deletes a key-value pair from the current in-session user's dictionary based on
+    * the "key" attribute of the DeleteRequest
+    * @param session HttpSession
+    * @param req DeleteRequest
+    * @return Response, 200 ok - payload: "Deleted",
+    * 401 unauthorized - payload: "Invalid session",
+    * 400 bad request - payload: ErrorResponse("Malformed JSON payload"),
+    * 422 Unprocessable Entity - payload: "no such entry" (if no entry exists for provided key)
+    */
+  @DELETE
+  @Path("/delete")
+  def deleteValue(@Session session: HttpSession, req: DeleteRequest): Response = {
+    val user = UserResource.getUser(session).orNull
+    if (user == null)
+      Response.status(Response.Status.UNAUTHORIZED).entity(ErrorResponse("Invalid session")).build()
+    else if (req == null || req.key == null) {
+      Response
+        .status(Response.Status.BAD_REQUEST)
+        .entity(ErrorResponse("Malformed JSON payload"))
+        .build()
+    } else if (!dictEntryExists(user, req.key)) {
+      Response.status(422).entity(ErrorResponse("No such entry")).build()
+    } else {
+      deleteDictEntry(user, req.key)
+      Response.ok(ConfirmationResponse("Deleted")).build()
+    }
+  }
+
+  /**
+    * This method retrieves a value from the user_dictionary table
+    * given a user's uid and key. each tuple (uid, key) is a primary key
+    * in user_dictionary, and should uniquely identify one value
+    */
+  private def getValueByKey(user: User, key: String): String = {
+    SqlServer.createDSLContext
+      .fetchOne(
+        USER_DICTIONARY,
+        USER_DICTIONARY.UID.eq(user.getUid).and(USER_DICTIONARY.KEY.eq(key))
+      )
+      .getValue
+  }
+
+  /**
+    * This method retrieves all of a user's dictionary entries in
+    * the user_dictionary table as a json object
+    */
+  private def getDict(user: User): Map[String, String] = {
+    Map(
+      asScalaBuffer(
```

Comment:
same, what is returned if the user does not exist in db?

Reply:
Returns an empty map, translating into an empty json object.

Reply:
Returns an empty map, which gets converted to an empty json object. I tested this with a user who had no dict entries.